### PR TITLE
[Web-UI] Add "Clear History" button and "Delete" buttons for old messages

### PR DESCRIPTION
**Pull request by keldenl**
Clearing out old messages and managing chat history can be a hassle, so I added convenient buttons to make it easier. In addition, I improved the styling of the buttons to better match their purpose.

Screenshot 2023-03-20 at 8 42 56 PM

### CODE REVIEW

File: bin/web/views/index.ejs [hunk 1]
```diff
@@ -1,246 +1,341 @@
 <html>
+
```
Comment (keldenl) [inline bin/web/views/index.ejs:2]: lots of whitespace changes unfortunately due to prettier, but i think the new spacing is more readable

File: bin/web/views/index.ejs [hunk 2]
```diff
@@ -1,246 +1,341 @@
...
+        
+        button {
+            padding: 10px;
+            background: rgba(0, 0, 0, 0.8);
+            color: white;
+            border: none;
+            border-radius: 3px;
+            outline: none;
+            box-sizing: border-box;
+            cursor: pointer;
```
Comment (keldenl) [inline bin/web/views/index.ejs:48]: hovering over buttons weren't giving the "cursor" look, so i added it

File: bin/web/views/index.ejs [hunk 3]
```diff
@@ -1,246 +1,341 @@
...
+            background: rgba(0, 0, 0, 0.15);
+        }
+        
+        .danger-btn {
+            background: rgba(255, 0, 0, 0.6);
+        }
+        
+        .danger-btn:hover {
+            background: rgba(255, 0, 0, 0.8);
+        }
```
Comment (keldenl) [inline bin/web/views/index.ejs:70]: added some new button styles (you can see it in the screenshot attached)

File: bin/web/views/index.ejs [hunk 4]
```diff
@@ -1,246 +1,341 @@
...
+                if (response == "\n\n<end>") {
+                    document.querySelector(".form-header").innerHTML = renderHeader(config)
+                } else {
+                    config.models.push(response)
+                }
+            } else {
+                const id = request.id
+                let existing = document.querySelector(`[data-id='${id}']`)
+                if (response == "\n\n<end>") {
+                    // Add a delete button
```
Comment (keldenl) [inline bin/web/views/index.ejs:319]: now adds an individual delete button whenever the generation is complete

**Comment by luluhoc** [comment]
https://github.com/cocktailpeanut/dalai/pull/136

You can do it with my frontend I'm waiting for PR to be accepted

**Comment by keldenl** [comment]
using another branch for this instead, closing this pr in favor this one https://github.com/cocktailpeanut/dalai/pull/202

**Pull request closed without merging**